image/types/src/lib.rs

```diff
@@ -392,4 +392,109 @@ impl ImageTocEntry {
     pub fn image_size(&self) -> u32 {
         self.size
     }
+
+    pub fn overlaps(&self, other: &ImageTocEntry) -> bool {
+        if self.load_addr < (other.load_addr + other.image_size())
```

Comment:
nitpick:

This if is unnecessary; this can just be

```
pub fn overlaps(&self, other: &ImageTocEntry) -> bool {
    self.load_addr < (other.load_addr + other.image_size())
    && (self.load_addr + self.image_size()) > other.load_addr
}
```

Reply:
Fixed.